transport/response.hh

```diff
@@ -141,6 +141,16 @@ private:
     }
 };
 
+inline std::unique_ptr<cql_server::response> make_unique_response(int16_t stream, cql_binary_opcode opcode, const tracing::trace_state_ptr& tr_state_ptr) {
```

Comment:
After some discussion we believe that the cleanest solution will be to set up the custom payload in the ctor of `response`. So we'd have one ctor: `response(int16_t, cql_binary_opcode, const tracing::trace_state_ptr&)` and we'd move the logic of this function into it.
That way we don't edit any unrelated code in `server.cc`, potentially avoiding merge conflicts in the future.